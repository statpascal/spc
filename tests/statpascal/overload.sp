program overloadtest;

procedure test (var a: integer);
    begin
	writeln ('var int ', a)
    end;

procedure test (a: integer);
    begin
	writeln ('integer ', a)
    end;

procedure test (a: real);
    begin
	writeln ('real ', a)
    end;

procedure test (a: int64vector);
    begin
        writeln ('intvec ', a)
    end;

procedure test (a: realvector);
    begin
  	writeln ('realvec ', a)
    end;

type 
    proc = procedure (a: real);

var
    p: proc;
    n: integer;

begin
    n := 3;
    test (2);
    test (2.5);
    test (n);
    test (intvec (1, 5));
    test (realvec (1, 2, 0.2));
    p := test;
    p (2.5)
end.
