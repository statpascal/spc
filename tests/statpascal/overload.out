integer 2
real 2.5
var int 3
intvec 1 2 3 4 5 
realvec 1 1.2 1.4 1.6 1.8 2 
real 2.5
